operator/v1/types_network.go

```diff
@@ -374,6 +373,11 @@ type OVNKubernetesConfig struct {
 	// reported defaults are used.
 	// +optional
 	PolicyAuditConfig *PolicyAuditConfig `json:"policyAuditConfig,omitempty"`
+	// gatewayConfig holds the configuration for node gateway options. If unset,
+	// reported defaults are used. Supported from OCP 4.10.
+	// +kubebuilder:default:={mode: shared}
```

Comment:
I'm not sure what "reported defaults are used" but I see you just copied that from other fields...

I don't know if we should be defaulting the value here; CNO is going to have to deal with `GatewayConfig` being `nil` in existing configs, so it seems like we should just leave it `nil` if unspecified in new configs as well.